node/src/main/scala/org/bitcoins/node/PeerManager.scala

```diff
@@ -260,7 +268,7 @@ case class PeerManager(
               .map { _ =>
                 //could have already been deleted in case of connection issues
                 if (finder.hasPeer(peer))
-                  finder.getData(peer).client.close()
+                  finder.getData(peer).client.foreach(_.close())
```

Comment:
Use `.map`, don't use `foreach` with `Future` as it encourages dropping `Future`'s on the floor.